guides/hack/11-built-in-types/71-dynamic-examples/coercion_from_union.php.type-errors.example.typechecker.out

```diff
@@ -0,0 +1,7 @@
+File "/home/example/coercion_from_union.php", line 16, characters 17-18:
```

Comment:
Could we use a different extension? In the Hack test suite itself, we have foo.php source code, foo.php.exp expected output, and we regenerated foo.php.out on every run.

My assumption was that .out files shouldn't be committed.

Reply:
To be precise, this is not an "out" file, this is an "example out" file. These are needed when the output is non-deterministic (like this one because it contains randomly generated/system specific file paths).

- When the output is deterministic, we just have an .expect file which is used for testing as well as in the rendered HTML guides.
- When the output is non-deterministic, we have an .expect**f** file for testing, and need another file to be placed in the rendered HTML output since we don't want `%s` etc. visible in the guides (hence the "example out" file, which is a new concept not part of the normal Hack test suite).

I used the same pattern we've been using before, see e.g. the example out files here: https://github.com/hhvm/user-documentation/tree/master/guides/hack/15-asynchronous-operations/22-generators-examples (but now it's possible to have both "example.hhvm.out" and "example.typechecker.out" depending on whether we want to show HHVM or Hack output in the guide).